middleware/http/client_test.go

```diff
@@ -102,5 +103,17 @@ func TestWrapClientW3C(t *testing.T) {
 	assert.Equal(t, "01", parts[3], "w3c trace-flags not is not correct")
 
 	// Assert TraceState
-	assert.Equal(t, "dd=s:1;t.dm:-1", tracestate)
+	assert.False(t, strings.Contains(tracestate, ","), "w3c tracestate contained multiple list-members, but we only expected 1")
+	parts = strings.Split(tracestate, "=")
+	require.Equal(t, 2, len(parts))
+	require.Equal(t, "dd", parts[0], "w3c tracestate list-member did not contain Datadog list-member")
+
+	// Assert Datadog-part of TraceState
+	parts = strings.Split(parts[1], ";")
+	sort.Strings(parts) // The Datadog-part of the Tracestate can be in any order.
+	require.Equal(t, 3, len(parts))
+	assert.Equal(t, "s:1", parts[0])
+	assert.Equal(t, "t.dm:-1", parts[1])
+	assert.True(t, strings.HasPrefix(parts[2], "t.tid:"))
+	assert.Equal(t, len("t.tid:65796a3f00000000"), len(parts[2]), "t.tid had invalid length.\nExample: %s\nGot:     %s", "t.tid:65796a3f00000000", parts[2])
```

Comment:
Should we say that the part after `:` is 128-bit instead?

Reply:
I am not sure how to achieve it though.

Reply:
I was contemplating doing less validation at all, as the list-members are opaque strings unique and not publicly documented by Datadog

Reply:
Looks like this actually comes from the w3c spec: https://www.w3.org/TR/trace-context/#trace-id

Implement in dd-trace-go, here: https://github.com/DataDog/dd-trace-go/pull/2335

Reply:
Talked offline about it. The conclusion was we will ignore vendor specific tracestate header, [see](https://www.w3.org/TR/trace-context/#tracestate-header) for tests.